File: split/showbox.pas
procedure showbox(p: halfword);
begin
  {236:}
depththreshold := eqtb[5288].int;
  breadthmax := eqtb[5287].int{:236}
;
  if breadthmax <= 0 then
    breadthmax := 5;
  if poolptr + depththreshold >= poolsize then
    depththreshold := poolsize - poolptr - 1;
  shownodelist(p);
  println;
end;

File: cpp/showbox.cpp
#include "showbox.h"
#include "shownodelist.h"
#include "println.h"

static int show_box_breadth(void) { return int_par(show_box_breadth_code); }
static int show_box_depth(void) { return int_par(show_box_depth_code); }

void showbox(halfword p)
{  
	depththreshold = show_box_depth();
	breadthmax = show_box_breadth();
	if (breadthmax <= 0)
		breadthmax = 5;
	if (poolptr+depththreshold >= poolsize)
		depththreshold = poolsize-poolptr-1;
	shownodelist(p);
	println();
}
